app/main/activity/bar-chart.directive.js

```diff
@@ -89,13 +47,87 @@ function ActivityBarChartController($scope, $translate, PostEndpoint, d3, _, Pos
 
         $scope.isLoading = true;
         PostEndpoint.stats(postQuery).$promise.then(function (results) {
-            $scope.options.chart.xAxis.axisLabel = $translate.instant($scope.groupByOptions[$scope.groupBy.value]);
-            if (results.totals[0]) {
-                results.totals[0].key = $scope.options.chart.yAxis.axisLabel;
-            }
             $scope.data = results.totals;
+            generateGraph();
             $scope.isLoading = false;
         });
     }
 
+    function getRandomColour() {
```

Comment:
I just smiled at the simplicity of this function :)

Something for the future (not this PR), but maybe let's look for a palette of colors to randomly pick from, for better visual results.